### ISSUE TITLE
Build and Debug Windows target in VS Code with Uno extension fails

### ISSUE THREAD

**Issue by AndyCW** [unoplatform/uno#15101]
### Current behavior

Right-click on the Windows project and select Build - fails with error:
`C:\Users\andy.wigley\.nuget\packages\microsoft.windowsappsdk\1.4.231008000\buildTransitive\Microsoft.WindowsAppSDK.SelfContained.targets(36,5): error : The platform 'AnyCPU' is not supported for Self Contained mode. [C:\Data\development\UnoApp1\UnoApp1\UnoApp1.Windows\UnoApp1.Windows.csproj]`

Maybe this is because the dotnet build command needs to have the 'Platform=x64' property set? Attempted to rectify by following steps documented here: https://platform.uno/docs/articles/features/vscode-advanced-mobile-debugging.html#customizing-tasksjson:
- As documented, selected `Tasks: Configure Task` from the Command palette
- Picked the **Uno: net7.0-windows10.0.19041.0 | Debug | win-x642** task to copy
- Note: This task already sets the "-p:Platform=x64" arg, so no changes are made

The following is added to my tasks.json:
```
    {
      "type": "Uno",
      "targetFramework": "net7.0-windows10.0.19041.0",
      "args": [
        "\"${workspaceFolder}\\UnoApp1.Windows\\UnoApp1.Windows.csproj\"",
        "-p:Platform=x64",
        "-p:WindowsAppSDKSelfContained=true",
        "-p:AppendTargetFrameworkToOutputPath=false",
        "-c:Debug",
        "-r:win-x64",
        "-clp:NoSummary",
        "-p:GenerateFullPaths=true",
        "-p:UnoForceSingleTFM=true",
        "-bl:\"${workspaceFolder}\\UnoApp1.Windows\\net7.0-windows10.0.19041.0-Debug-win-x64.binlog\""
      ],
      "problemMatcher": [
        "$msCompile"
      ],
      "label": "Uno: net7.0-windows10.0.19041.0 | Debug | win-x64"
    }
```

Now when I try to build the project, I get a popup with message "There are task errors. See the output for details."
In the output window, there is:

```
Error: The Uno task detection didn't contribute a task for the following configuration:
{
    "type": "Uno",
    "targetFramework": "net7.0-windows10.0.19041.0",
    "args": [
        "\"${workspaceFolder}\\UnoApp1.Windows\\UnoApp1.Windows.csproj\"",
        "-p:Platform=x64",
        "-p:WindowsAppSDKSelfContained=true",
        "-p:AppendTargetFrameworkToOutputPath=false",
        "-c:Debug",
        "-r:win-x64",
        "-clp:NoSummary",
        "-p:GenerateFullPaths=true",
        "-p:UnoForceSingleTFM=true",
        "-bl:\"${workspaceFolder}\\UnoApp1.Windows\\net7.0-windows10.0.19041.0-Debug-win-x64.binlog\""
    ],
    "problemMatcher": [
        "$msCompile"
    ],
    "label": "Uno: net7.0-windows10.0.19041.0 | Debug | win-x64"
}
The task will be ignored.
```

### Expected behavior

The windows app should build successfully using the Tasks supplied in the Uno Platform extension.
It should not be necessary to create a custom task.

### How to reproduce it (as minimally and precisely as possible)

[UnoApp1.zip](https://github.com/unoplatform/uno/files/13966210/UnoApp1.zip)

Reproduces with a minimal Uno app created in the wizard (as attached)

Steps:

1. On Windows, Using VS Code with the C# dev kit extension, and Uno extension installed, open the attached project
2. In Solution Explorer, right-click on the UnoApp1.Windows project and then click Build - build fails

### Workaround

Can build at the command line from the UnoApp1.Windows folder, using:
`dotnet build -f net7.0-windows10.0.19041.0 -r win-x64 -c Debug -p win-x64`

### Works on UWP/WinUI

Yes

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia

### NuGet package version(s)

<Project ToolsVersion="15.0">
  <ItemGroup>
    <PackageVersion Include="Microsoft.Extensions.Logging.Console" Version="7.0.0" />
    <PackageVersion Include="Microsoft.Windows.SDK.BuildTools" Version="10.0.22621.756" />
    <PackageVersion Include="Microsoft.WindowsAppSDK" Version="1.4.231008000" />
    <PackageVersion Include="SkiaSharp.Skottie" Version="2.88.6" />
    <PackageVersion Include="SkiaSharp.Views.Uno.WinUI" Version="2.88.6" />
    <PackageVersion Include="Uno.Core.Extensions.Logging.Singleton" Version="4.0.1" />
    <PackageVersion Include="Uno.Extensions.Logging.OSLog" Version="1.7.0" />
    <PackageVersion Include="Uno.Resizetizer" Version="1.2.0" />
    <PackageVersion Include="Uno.UI.Adapter.Microsoft.Extensions.Logging" Version="5.0.48" />
    <PackageVersion Include="Uno.UniversalImageLoader" Version="1.9.36" />
    <PackageVersion Include="Uno.WinUI" Version="5.0.48" />
    <PackageVersion Include="Uno.WinUI.Lottie" Version="5.0.48" />
    <PackageVersion Include="Uno.WinUI.DevServer" Version="5.0.48" />
    <PackageVersion Include="Uno.WinUI.Skia.Gtk" Version="5.0.48" />
  </ItemGroup>
</Project>

### Affected platforms

Windows

### IDE

Visual Studio Code

### IDE version

1.85.1

### Relevant plugins

C# dev kit extension
Uno Platform extension

**Comment by jeromelaban** [comment]
Thanks for the report! 

At this time, debugging in VS Code does not work properly because of some missing compatibility of WinAppSDK with `dotnet build`. There are are workarounds possible (similar to https://github.com/unoplatform/uno/discussions/12471 and the one you found), but those are causing other issues that are difficult to diagnose.

It may be that the newest update 1.4 of WinAppSDK adjusts this behavior and will allow for debugging to work, @spouliot may know more.

**Comment by spouliot** [comment]
I made some progress before the holidays but I got stuck on a on a different _build_ (not debugger) issue. Some files are not at the expected location and this makes the Uno app throws an unhandled exception (which the debugger can break into).

**Comment by spouliot** [comment]
Note that your suggested workaround does not work for me.

```
C:\Users\sebas\Downloads\UnoApp1\UnoApp1\UnoApp1.Windows>dotnet build -f net7.0-windows10.0.19041.0 -r win-x64 -c Debug -p win-x64
MSBuild version 17.8.3+195e7f5a3 for .NET
MSBuild logs and debug information will be at "C:\Users\sebas\Downloads\UnoApp1\UnoApp1\UnoApp1.Windows\MSBuild_Logs"
  Determining projects to restore...
  All projects are up-to-date for restore.
MSBuild logs and debug information will be at "C:\Users\sebas\Downloads\UnoApp1\UnoApp1\UnoApp1.Windows\MSBuild_Logs"
C:\Users\sebas\.nuget\packages\microsoft.windowsappsdk\1.4.231008000\buildTransitive\Microsoft.WindowsAppSDK.SelfContained.targets(3
6,5): error : The platform 'AnyCPU' is not supported for Self Contained mode. [C:\Users\sebas\Downloads\UnoApp1\UnoApp1\UnoApp1.Wind
ows\UnoApp1.Windows.csproj]

Build FAILED.

C:\Users\sebas\.nuget\packages\microsoft.windowsappsdk\1.4.231008000\buildTransitive\Microsoft.WindowsAppSDK.SelfContained.targets(3
6,5): error : The platform 'AnyCPU' is not supported for Self Contained mode. [C:\Users\sebas\Downloads\UnoApp1\UnoApp1\UnoApp1.Wind
ows\UnoApp1.Windows.csproj]
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:00.23
```

However changing the last parameter on the command-line, like below, does build.

```
dotnet build -f net7.0-windows10.0.19041.0 -r win-x64 -c Debug -p Platform=x64
```

but the produced `.exe` does not work :( which seems to be the same problem I was confronted last time.

**Comment by AndyCW** [comment]
Yes, sorry, my "workaround" was wrong. 
I can confirm what you are saying, with that command it does build, but the resulting exe won't run.

**Comment by spouliot** [comment]
With @nickrandolph help I was able to build the app _properly_ and debugging started to work as expected.

The changes require a new release of the VS Code extension.

**Comment by spouliot** [comment]
Support to build and debug WinUI apps was released with the Uno extension 0.13.
